server/src/main/java/org/opensearch/transport/TransportResponseHandler.java

```diff
@@ -52,6 +52,8 @@ public interface TransportResponseHandler<T extends TransportResponse> extends W
 
     String executor();
 
+    default void handleRejection(Exception exp) {};
```

Comment:
Please add javadoc to explain why we added it and under what condition it will be called